e2e/mock/server.go

```diff
@@ -1,41 +1,59 @@
 package mock
 
 import (
-	"net"
-
-	"github.com/hyperledger/fabric-protos-go/orderer"
-	"github.com/hyperledger/fabric-protos-go/peer"
-	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
 type Server struct {
-	GrpcServer *grpc.Server
-	Listener   net.Listener
+	peers   []*Peer
+	orderer *Orderer
 }
 
-func (s *Server) Start() {
+func NewServer(peerN int, Credentials credentials.TransportCredentials) (*Server, error) {
 	blockC := make(chan struct{}, 1000)
+	peers := make([]*Peer, 0)
+
+	for i := 0; i < peerN; i++ {
+		peer, err := NewPeer(blockC, Credentials)
+		if err != nil {
+			return nil, err
+		}
+		peers = append(peers, peer)
+	}
 
-	p := &Peer{
-		BlkSize: 10,
-		TxC:     blockC,
+	orderer, err := NewOrderer(blockC, Credentials)
+	if err != nil {
+		return nil, err
 	}
+	return &Server{peers: peers, orderer: orderer}, nil
+}
 
-	o := &Orderer{
-		TxC: blockC,
+func (s *Server) Start() {
+	for _, v := range s.peers {
+		go v.Start()
 	}
+	go s.orderer.Start()
+}
 
-	peer.RegisterEndorserServer(s.GrpcServer, p)
-	peer.RegisterDeliverServer(s.GrpcServer, p)
-	orderer.RegisterAtomicBroadcastServer(s.GrpcServer, o)
+func (s *Server) Stop() {
+	for _, v := range s.peers {
+		v.Stop()
+	}
+	s.orderer.Stop()
+}
 
-	err := s.GrpcServer.Serve(s.Listener)
-	if err != nil {
-		panic(err)
+func (s *Server) GetPeersAddresses() (peersAddrs []string) {
+	peersAddrs = make([]string, len(s.peers))
+	for k, v := range s.peers {
+		peersAddrs[k] = v.getAdders()
 	}
+	return
 }
 
-func (s *Server) Stop() {
-	s.GrpcServer.Stop()
-	s.Listener.Close()
+func (s *Server) GetOrderAddr() string {
+	return s.orderer.getAdders()
+}
+
+func (s *Server) GetAdders() ([]string, string) {
+	return s.GetPeersAddresses(), s.GetOrderAddr()
 }
```

Comment:
conventionally in Go, getters do *not* start with `Get`, for example:
```go
func (z *Zoo) Cats() []Cat {}
func (z *Zoo) SetCats(cats []Cat) {}
```

Reply:
if you add `GetAddrs`, it should be used in favor of `GetPeerAddresses` and `GetOrdererAddr`